apps/src/lab2/views/components/Instructions.tsx

```diff
@@ -37,13 +68,28 @@ const Instructions = ({baseUrl, vertical, right}) => {
   );
 };
 
-Instructions.propTypes = {
-  baseUrl: PropTypes.string.isRequired,
-  vertical: PropTypes.bool,
-  right: PropTypes.bool,
-};
+interface InstructionsPanelProps {
+  /** Primary instructions text to display. */
+  text: string;
+  /** Optional message to display under the main text. This is typically a validation message. */
+  message?: string;
+  /** Optional image URL to display. */
+  imageUrl?: string;
+  /** If the next button should be shown. */
+  showNextButton?: boolean;
+  /** Callback to call when clicking the next button. */
+  onNextPanel?: () => void;
+  /** If the instructions panel should be rendered vertically. Defaults to false */
+  vertical?: boolean;
+  /** If the instructions panel should be rendered on the right side of the screen. Defaults to false. */
+  right?: boolean;
```

Comment:
It feels like the host app should know about these, and the instructions panel itself should be agnostic to its screen placement.  This philosophy is also helpful when we think about responsibility for panel layout, positioning, and size management, which should not be done by an individual panel.

Reply:
That seems reasonable.  We could also consider removing image pop-out support since we haven't used it for a while, and revive it later if we need it again.

Reply:
Agreed. Filed https://codedotorg.atlassian.net/browse/SL-1074 to update these.